numba_rvsdg/core/datastructures/basic_block.py

```diff
@@ -103,13 +118,21 @@ class SyntheticBranch(SyntheticBlock):
     branch_value_table: dict = None
 
     def replace_jump_targets(self, jump_targets: Tuple) -> "BasicBlock":
-        fallthrough = len(jump_targets) == 1
+        """Replaces jump targets of this block by the given tuple.
+        The provided jump targets must be in the same order as their
+        intended original replacements. Additionally also updates the
+        branch value table of the branch block.
+        
+        Note that replacing jump targets will not replace the backedge
```

Comment:
Could this be enforced programatically. I.e. When we replace the the jump_targets, we can replace the declared back-edges too? Since the new jump_targets are in order, we could determine which ones are be replaced by which others ones and such do the update?

Reply:
One thing that I was thinking regarding this (and maybe we'd want to make this change in 0.0.3) is `back_edges` can simply be a tuple of booleans stating whether the `_jump_target` in that position is a backedge or not. This is equivalent to the `back_edge` tuple saying: true if the jump target at position x is a backedge and false if it's not.

This removes a lot of instances where we have to 'update' back edges just because the jump target was changed. Unless the position of the back edge is changing which in the majority of cases is not possible at all, we wouldn't need to touch the `back_edge` variable at all.

Reply:
@kc611 ok, can you open an issue regarding this, such that we don't forget. We can leave this as is for now and change it for 0.0.3.

Reply:
created it: https://github.com/numba/numba-rvsdg/issues/66